ocaml/xapi/xapi_vdi.ml

```diff
@@ -758,4 +758,90 @@ let read_database_pool_uuid ~__context ~self =
 	| Some (_, uuid) -> uuid
 	| None -> ""
 
+(* Detach the rrd-stats vdi *)
+let detach_rrd_vdi ~__context ~vbd =
+	Helpers.call_api_functions ~__context (fun rpc session_id ->
+		Client.VBD.unplug ~rpc ~session_id ~self:vbd);
+	Helpers.call_api_functions ~__context (fun rpc session_id ->
+		Client.VBD.destroy ~rpc ~session_id ~self:vbd)
+
+(* Set the unsigned int at an offset *)
+let set_cstruct_int cstruct offset num =
+	Cstruct.BE.set_uint32 cstruct offset num
+
+(* Get the unsigned int from an offset *)
+let get_cstruct_int cstruct offset =
+	Cstruct.BE.get_uint32 cstruct offset
+
+(* Read the string from an offset of specified length *)
+let read_string cstruct offset len =
+	let curr_text = String.make len '\000' in
+	Cstruct.blit_to_string cstruct offset curr_text 0 len;
+	curr_text
+
+(* Write the string to an offset of specified length *)
+let write_string cstruct offset text len =
+	Cstruct.blit_from_string text 0 cstruct offset len
+
+let format_rrd_vdi ~cstruct =
+	let magic_number = (0x7ada7adal) and version = 1l in
```

Comment:
Magic number provided in the design http://xapi-project.github.io/rrdd/futures/sr-level-rrds.html is altered here.
In design it's 0xda7ada7a, Here it's 0x7ada7ada.
Reason: It's overflowing a 32 bit number, it's safe to use a smaller number.